packages/quantic/playwright/page-object/analytics.ts

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Page, Request, expect} from '@playwright/test';
+import {
+  isEventProtocol,
+  isUaClickEvent,
+  isUaCustomEvent,
+} from '../utils/requests';
+
+export class AnalyticsHelper {
+  constructor(
+    protected page: Page,
+    protected trackingId: string
+  ) {
+    this.page = page;
+    this.trackingId = trackingId;
+  }
+
+  static async setCookieToEnableNextAnalytics(
+    page: Page,
+    url: string,
+    trackingId: string
+  ) {
+    const context = page.context();
+    await context.addCookies([
+      {name: 'LSKey-c$Coveo-Pendragon', value: trackingId, url},
+      {name: 'Coveo-Pendragon', value: trackingId, url},
+    ]);
+  }
+
+  async waitForClickUaAnalytics(
```

Comment:
what about ua search events?

Reply:
I would add the search logic when it's actually being used, so I would defer it in a next PR when we can start using it and confirming that it's really fulfilling the need with another component that triggers search.